react/README.md

```diff
@@ -450,6 +450,7 @@
     ```
 
   - Do not use underscore prefix for internal methods of a React component.
+  > Why? Underscore prefixes are sometimes used as a convention in other languages to denote privacy. But, unlike those languages, there is no native support for privacy in JavaScript, everything is public. Regardless of your intentions,  adding underscore prefixes to your properties does not actually make them private, and any property (underscore prefixed or not) should be treated as being public. See issues #1024, and #490 for a more in-depth discussion.
```

Comment:
I'm only aware of the convention in three other languages (there may be more):

**Python**
> _single_leading_underscore : weak "internal use" indicator. E.g. from M import * does not import objects whose name starts with an underscore. - [PEP8 StyleGuide](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles)

**C#**
I'm not an ex-C# programmer, but my colleagues are, and they tell me this is where their use of the underscore prefixes to denote privacy came from.
> We use _camelCase for internal and private fields and use readonly where possible. Prefix instance fields with _, static fields with s_ and thread static fields with t_. When used on static fields, readonly should come after static (i.e. static readonly not readonly static). - [.NET team styleguide](https://github.com/dotnet/corefx/blob/368fdfd86ee3a3bf1bca2a6c339ee590f3d6505d/Documentation/coding-guidelines/coding-style.md)

**Objective-C**
> Make sure the name of the instance variable concisely describes the attribute stored. Usually, you should not access instance variables directly, instead you should use accessor methods (you do access instance variables directly in init and dealloc methods). To help to signal this, prefix instance variable names with an underscore (_), for example - [Apple Guidelines for Cocoa?](http://stackoverflow.com/questions/3521254/prefixing-property-names-with-an-underscore-in-objective-c)